File: src/main/java/cses/introductory_problems/IncreasingArray.java
package cses.introductory_problems;

import java.io.*;
import java.util.StringTokenizer;

public class IncreasingArray {
    static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));

        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());

        long previous = Long.parseLong(st.nextToken());
        long result = 0;

        for (int i = 1; i < n; i++) {
            long number = Long.parseLong(st.nextToken());
            if (number < previous) {
                result += previous - number;
            } else {
                previous = number;
            }
        }

        out.println(result);
        out.flush();
    }
}

File: src/main/kotlin/cses/introductory_problems/IncreasingArray.kt
package introductory_problems

fun main() {
//    val n = readln().toInt()
//    val numbers = readln().split(" ").map { it.toInt() }

    val n = 5
    val numbers = listOf(3, 2, 5, 1, 7)

    val answer = solution(n, numbers)

    println(answer)
}

fun solution(n: Int, numbers: List<Int>): Int {
    var result = 0
    var previous = numbers.first()

    for (i in 1 until n) {
        val number = numbers[i]
        if (number < previous) {
            result += previous - number
        } else if (number > previous) previous = number
    }

    return result
}